include/fixed_containers/fixed_bitset.hpp

```diff
@@ -0,0 +1,672 @@
+// * memcmp in FixedBitSet::operator==
+// * reinterpret_cast in FixedBitSet::count
+//
+// Original code from https://github.com/neargye-wg21/bitset-constexpr-proposal
+
+#include "fixed_containers/assert_or_abort.hpp"
+#include "fixed_containers/sequence_container_checking.hpp"
+
+#include <array>
+#include <bit>
+#include <cassert>
+#include <climits>
+#include <cstddef>
+#include <functional>
+#include <string>
+#include <type_traits>
+
+namespace fixed_containers
+{
+
+template <size_t BIT_COUNT,
+          customize::SequenceContainerChecking CheckingType =
+              customize::SequenceContainerAbortChecking<bool, BIT_COUNT>>
+class FixedBitset
+{  // store fixed-length sequence of Boolean elements
+
+private:
+    using UnsignedLong = unsigned long;           // NOLINT(google-runtime-int)
+    using UnsignedLongLong = unsigned long long;  // NOLINT(google-runtime-int)
+
+public:
+    using size_type = std::size_t;
+    using difference_type = std::ptrdiff_t;
+    using Checking = CheckingType;
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable : 4296)  // expression is always true (/Wall)
+#endif
+
+    using Ty = std::
+        conditional_t<BIT_COUNT <= sizeof(UnsignedLong) * CHAR_BIT, UnsignedLong, UnsignedLongLong>;
+
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+
+    class Reference
+    {  // proxy for an element
+        friend FixedBitset<BIT_COUNT>;
+
+    public:
+        constexpr ~Reference() noexcept = default;
+
+        constexpr Reference& operator=(bool val) noexcept
+        {
+            p_bitset_->set_unchecked(my_pos_, val);
+            return *this;
+        }
+
+        constexpr Reference& operator=(const Reference& bitref) noexcept
+        {
+            if (this == &bitref)
+            {
+                return *this;
+            }
+
+            p_bitset_->set_unchecked(my_pos_, static_cast<bool>(bitref));
+            return *this;
+        }
+
+        constexpr Reference& flip() noexcept
+        {
+            p_bitset_->flip_unchecked(my_pos_);
+            return *this;
+        }
+
+        constexpr bool operator~() const noexcept { return !p_bitset_->subscript(my_pos_); }
+
+        constexpr operator bool() const noexcept { return p_bitset_->subscript(my_pos_); }
+
+    private:
+        constexpr Reference() noexcept
+          : p_bitset_(nullptr)
+          , my_pos_(0)
+        {
+        }
+
+        constexpr Reference(FixedBitset<BIT_COUNT>& bitset, size_t pos)
+          : p_bitset_(&bitset)
+          , my_pos_(pos)
+        {
+        }
+
+        FixedBitset<BIT_COUNT>* p_bitset_;
+        size_t my_pos_;  // position of element in FixedBitSet
+    };
+
+    using reference = Reference;
+
+    static constexpr void validate(size_t pos)
+    {  // verify that _Pos is within bounds
+        assert_or_abort(pos < BIT_COUNT && static_cast<bool>("FixedBitSet index outside range"));
+    }
+
+    [[nodiscard]] constexpr bool subscript(size_t pos) const
```

Comment:
I think this should be moved to a helper that is named `subscript_unchecked` to match the `set_unchecked` naming and not confuse me while reading the code.

Reply:
Done.